### PR TITLE
Move capv to v1alpha3

### PR DESCRIPTION
**Pull request by yastij**
**What this PR does / why we need it**: This PR moves CAPV types + controllers to v1alpha3 types. This PR also adds support for `ControlPlaneEndpoint` and removes reporting API Endpoint through `.Status.APIEndpoints`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Move CAPV to v1alpha3
```

### CODE REVIEW

File: controllers/vspherecluster_controller.go [hunk 1]
```diff
@@ -215,7 +215,7 @@ func (r clusterReconciler) reconcileNormal(ctx *context.ClusterContext) (reconci
 
 func (r clusterReconciler) reconcileAPIEndpoints(ctx *context.ClusterContext) error {
 	// If the cluster already has API endpoints set then there is nothing to do.
-	if len(ctx.VSphereCluster.Status.APIEndpoints) > 0 {
+	if ctx.VSphereCluster.Spec.ControlPlaneEndpoint != (infrav1.APIEndpoint{}) {
```
Comment (yastij) [inline controllers/vspherecluster_controller.go]: This shouldn't be required anymore, but can be addressed in a follow-up PR

**Comment by akutz** [comment]
I'm merging this as is since the e2e tests will not pass until we introduce the new, v1alpha3-compatible test framework.

**Pull request merged**